libraries/fwd/fwd_bem_model.cpp

```diff
@@ -321,13 +327,37 @@ float **mne_mat_mat_mult_40 (float **m1,float **m2,int d1,int d2,int d3)
     int j,k,p;
     float sum;
 
-    for (j = 0; j < d1; j++)
+    // TODO: Hack to include faster Eigen processing. This should eventually be replaced dwith pure Eigen logic.
+    MatrixXf a(d1,d2);
+    MatrixXf b(d2,d3);
+
+    for (j = 0; j < d1; j++) {
+        for (k = 0; k < d2; k++) {
+            a(j,k) = m1[j][k];
+        }
+    }
+
+    for (j = 0; j < d2; j++) {
+        for (k = 0; k < d3; k++) {
+            b(j,k) = m2[j][k];
+        }
+    }
+
+    MatrixXf resultMat = a * b;
+
+    for (j = 0; j < d1; j++) {
         for (k = 0; k < d3; k++) {
-            sum = 0.0;
-            for (p = 0; p < d2; p++)
-                sum = sum + m1[j][p]*m2[p][k];
-            result[j][k] = sum;
+            result[j][k] = resultMat(j,k);
         }
+    }
+
+//    for (j = 0; j < d1; j++)
+//        for (k = 0; k < d3; k++) {
+//            sum = 0.0;
+//            for (p = 0; p < d2; p++)
+//                sum = sum + m1[j][p]*m2[p][k];
+//            result[j][k] = sum;
+//        }
```

Comment:
has this been tested?

Reply:
yes. the test returns the same results for this hack.